src/Build/BuildCheck/Infrastructure/CustomConfigurationData.cs

```diff
@@ -15,17 +15,27 @@ namespace Microsoft.Build.Experimental.BuildCheck;
 ///  that were attribute to a particular rule, but were not recognized by the infrastructure.
 /// The configuration data that is recognized by the infrastructure is passed as <see cref="BuildAnalyzerConfiguration"/>.
 /// </summary>
-/// <param name="ruleId"></param>
-public class CustomConfigurationData(string ruleId)
+public class CustomConfigurationData
```

Comment:
nit: Is this class intended to be derived from? If not, it should be `sealed` and the `Equals` implementation can use `is` instead of `obj.GetType() != this.GetType()`.

Reply:
Not it is not supposed to be derived (at this point at least),
Updated the usage thank you